network/wsNetwork.go

```diff
@@ -661,6 +687,9 @@ func (wn *WebsocketNetwork) setup() {
 	if wn.config.DNSSecurityRelayAddrEnforced() {
 		preferredResolver = dnssec.MakeDefaultDnssecResolver(wn.config.FallbackDNSResolverAddress, wn.log)
 	}
+	if wn.nodeInfo == nil {
+		wn.nodeInfo = &nopeNodeInfo{}
```

Comment:
since this code path is really only used by algodump, netdump and a test (callers to NewWebsocketGossipNode) maybe we should just leave it nil (and update OnNetworkAdvance to check nodeInfo == nil and exit).